netbox/dcim/forms.py

```diff
@@ -2725,6 +2725,23 @@ class Meta:
             'termination_b_site', 'termination_b_rack', 'termination_b_device', 'termination_b_id', 'type', 'status',
             'label', 'color', 'length', 'length_unit',
         ]
+        widgets = {
+            'type': StaticSelect2(),
+            'status': StaticSelect2(),
+            'length_unit': StaticSelect2(),
+        }
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        # Initialize site and rack fields to match that of the existing temination
+        if self.instance and self.instance.termination_a:
+            termination_a_parent = self.instance.termination_a.parent
+            if hasattr(termination_a_parent, 'site'):
+                self.initial['termination_b_site'] = termination_a_parent.site
```

Comment:
Be careful with initial data. As written, this will override any initial data passed as a URL query parameter (e.g. `?termination_b_site=123`). We only want to set `self.initial['termination_b_site']` if it doesn't already exist.

Reply:
Thanks for pointing that out. I've fixed this and will keep it in mind in the future; I've never touched Django before in my life, so I appreciate your patience with my rookie mistakes.